components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -170,288 +166,187 @@ subroutine ma_convproc_intr( state, ptend, pbuf, ztodt,             &
+   real(r8), intent(in)    :: dp(pcols,pver)    ! Delta pressure between interfaces [mb]
+   integer,  intent(in)    :: jt(pcols)         ! Index of cloud top (updraft top) for each column in w grid
+   integer,  intent(in)    :: maxg(pcols)       ! Index of cloud base (level of maximum moist static energy) for each column in w grid
    integer,  intent(in)    :: ideep(pcols)      ! Gathering array
    integer,  intent(in)    :: lengath           ! Gathered min lon indices over which to operate
-   integer,  intent(in)    :: species_class(:)
-   logical,  intent(in)    :: history_aero_prevap_resusp
+   integer,  intent(in)    :: species_class(:)  ! species index
 
 
 ! Local variables
-   integer, parameter :: nsrflx = 6        ! last dimension of qsrflx
-   integer  :: i, ii 
-   integer  :: k
-   integer  :: l, ll, lchnk
-   integer  :: n, ncol, nstep
-
-   real(r8) :: dlfdp(pcols,pver)
-   real(r8) :: dpdry(pcols,pver)
-   real(r8) :: dqdt(pcols,pver,pcnst)
-   real(r8) :: dt
-   real(r8) :: qa(pcols,pver,pcnst), qb(pcols,pver,pcnst)
-   real(r8) :: qsrflx(pcols,pcnst,nsrflx)
-   real(r8) :: sflxic(pcols,pcnst)
-   real(r8) :: sflxid(pcols,pcnst)
-   real(r8) :: sflxec(pcols,pcnst)
-   real(r8) :: sflxed(pcols,pcnst)
-   real(r8) :: tmpa, tmpb, tmpg
-
-   logical  :: dotend(pcnst)
-
-! physics buffer fields 
-   integer itim, ifld
-   real(r8), pointer, dimension(:,:,:) :: fracis  ! fraction of transported species that are insoluble
+   integer  :: ncol             ! total column number. from state%ncol
+   logical  :: dotend(pcnst)    ! if do tendency
+
+   real(r8) :: dlfdp(pcols,pver)                ! Deep (Total-Shallow) conv cldwtr detrainment (grid avg) [kg/kg/s] 
+   real(r8) :: dqdt(pcols,pver,pcnst)           ! time tendency of q [kg/kg/s]
+   real(r8) :: qnew(pcols,pver,pcnst)           ! tracer mixing ratio from state%q [kg/kg]
+                                                ! qnew is updated through the processes in this subroutine but does not update into state
+
+   integer, parameter :: nsrflx = 6             ! last dimension of qsrflx
+   real(r8) :: qsrflx(pcols,pcnst,nsrflx)       ! process-specific column tracer tendencies
+                            !  1 = activation   of interstial to  conv-cloudborne
+                            !  2 = resuspension of conv-cloudborne to interstital
+                            !  3 = aqueous chemistry (not implemented yet, so  zero)
+                            !  4 = wet removal
+                            !  5 = actual precip-evap resuspension (what actually is applied to a species)
+                            !  6 = pseudo precip-evap resuspension (for history file)
+
 
 !
 ! Initialize
 !
-
-   lchnk = state%lchnk
-   ncol  = state%ncol
-   nstep = get_nstep()
-   dt = ztodt
-
-   hund_ovr_g = 100.0_r8/gravit
+  ncol  = state%ncol
+  qnew(1:ncol,:,:) = state%q(1:ncol,:,:)
+  dotend(:) = ptend%lq(:)
+  dqdt(:,:,:) = ptend%q(:,:,:)
+  hund_ovr_g = 100.0_r8/gravit
```

Comment:
Can this be a parameter? I am not sure if gravit is declared as a parameter or not.